io-engine/src/bdev/nexus/nexus_channel.rs

```diff
@@ -218,13 +247,45 @@ impl<'n> NexusChannel<'n> {
             }
         }
 
+        // add all active rebuild logs.
```

Comment:
Is it possible to cleanly merge this block with above block. The invariant being:
is_logging() implies is_rebuilding() but not other way round?

Reply:
Probably yes, I'll check.

Reply:
This could be rewritten in a single "for" block in "imperative" style, but here we have kinda "functional" style which utilizes ideas of iterators, filters and maps. With functional style, you are often forced to do one task at a time.

Generally, for non-async code, functional style is preferred in Rust, similarly to Python comprehensions. It may require multiple re-iterations in certain cases like here, but for non-CPU critical code it is okay.

The benefit of functional style is clearer separation of logic and tasks (once you get used to it, initially it can be intimidating, especially if language does not provide special functional syntax, like in Python or Haskell).

Also, in certain situations, Rust's borrow checker prevents an easy use of "for" loops other mutable collections.